lib/application.js

```diff
@@ -225,9 +224,10 @@ function respond(ctx) {
     return res.end();
   }
 
-  if ('HEAD' == ctx.method) {
-    if (!res.headersSent && isJSON(body)) {
-      ctx.length = Buffer.byteLength(JSON.stringify(body));
+  if ('HEAD' === ctx.method) {
+    if (!res.headersSent && !ctx.response.has('Content-Length')) {
+      const { length } = ctx.response;
+      if (Number.isInteger(length)) { ctx.length = length; }
```

Comment:
I'd like to remove `{}` in single line, others LGTM

Reply:
Fixed. Also calling `.hasHeader` directly on `res` here to make the logic here clearer and be in line with `res.headersSent` check in the same condition. 
Btw, did the team discussed the introduction of [Prettier](https://prettier.io/docs/en/why-prettier.html) on codebase to avoid code style reviews? I can make a PR for it 🙋🏻‍♂️

Reply:
I would prefer [Koa eslint config](https://github.com/koajs/eslint-config-koa) to be updated if anything.